src/combinators/mod.rs

```diff
@@ -1,35 +1,46 @@
 use gen::GenError;
 use lib::std::io::Write;
+use lib::std::io;
 
-pub type GenResult<I> = Result<(I, usize), GenError>;
+pub type GenResult<I> = Result<I, GenError>;
 
 pub trait SerializeFn<I>: Fn(I) -> GenResult<I> {}
 
 impl<I, F: Fn(I) -> GenResult<I>> SerializeFn<I> for F {}
 
-pub trait GenChain<I, S: SerializeFn<I>> {
-    fn chain(self, s: &S) -> GenResult<I>;
+pub trait Skip {
```

Comment:
Why are you dropping GenChain?

Reply:
Because it's useless/redundant :) It's the same as `Result::and_then` now, and apart from that it's a weird function to have in here. It's not a combinator for serializers (that would be `pair`, for example) but a combinator for `Result`s. And for that `std` already has all we need here.

For combining combinators you'd use `pair` (or `tuple`, which I'm going to add next here once this is in), for combining the results of serializers you'd use the `Result` combinators or plain pattern matching, `?` operator, etc.

For example for the `pair` combinator:

```rust
first(out).and_then(&second)
```

is the same as

```rust
first(out).chain(&second)
```

Reply:
Fair enough